File: ios/Runner/pdf/PdfRenderFrame.swift
import Foundation
import UIKit

struct PdfRenderFrame {
  let pageIndex: Int
  let pageCount: Int
  let baseImage: UIImage?
  let strokes: [StrokeOperation]
  let highlights: [HighlightOperation]
  let texts: [TextOperation]
  let images: [ImageOperation]
}

File: android/app/src/main/kotlin/com/pixeldev/Docly/pdf/PdfRenderFrame.kt
package com.pixeldev.Docly.pdf

import android.graphics.Bitmap

data class PdfRenderFrame(
    val pageIndex: Int,
    val pageCount: Int,
    val baseBitmap: Bitmap?,
    val strokes: List<StrokeOperation>,
    val highlights: List<HighlightOperation>,
    val texts: List<TextOperation>,
    val images: List<ImageOperation>,
)
